apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +124,54 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp calculate_package_total(
+         %{tax_total: tax_total, adjustment_total: adjustment_total, promo_total: promo_total},
+         shipping_cost
+       ) do
+    Enum.reduce([tax_total, adjustment_total, promo_total, shipping_cost], &Money.add!/2)
+  end
+
+  def process_package(package) do
```

Comment:
IMO this should be a private function.
This function assumes that the `package.shipping_method_id` is already set, see L138.
Then the same `shipping_method.id` is set as the `shipping_method_id` in L148. :confused: 

Since this PR moves us to `delivery` state (clearly defined in the [meeting](https://hackmd.io/atLBxkpvReiEwSbx5CzUTA#Live-coding) and [here](https://hackmd.io/4Neif6WUQIGSftYPpiMMTw?view#States-of-an-Order)), we should be "setting" the various package total fields (and not just reading them).
You are summing the package totals, but we know for a fact that `tax_total`, `adjustment_total` and `promo_total` are `nil`s.